DSA/stack.cpp

```diff
@@ -0,0 +1,215 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+// Creating Node Class
+
+class Node
+{
+public:
+    int data;
+    Node *next;
+
+    Node(int data)
+    {
+        this->data = data;
+        next = NULL;
+    }
+};
+
+// function to print meassage
+
+void print(string message)
+{
+    cout << endl
+         << message;
+}
+
+// Creating stack class with its functions
+
+class Stack
+{
+public:
+    Node *top;
+
+    Stack()
+    {
+        top = NULL;
+    }
+
+    // to push
+
+    void push()
+    {
+
+        // Taking input
+        print("Enter the data: ");
+        int data;
+        cin >> data;
+
+        Node *newNode = new Node(data);
+
+        if (!newNode)
+        {
+            print("Stack Overflow.");
+            return;
+        }
+
+        newNode->next = top;
+        top = newNode;
+    }
+
+    // to peek
+
+    void peek()
+    {
+        if (top == NULL)
+        {
+            print("Stack is empty.");
+        }
+        else
+        {
+            cout << endl
+                 << "The element is : " << this->top->data;
+        }
+    }
+
+    // tot pop
+
+    void pop()
+    {
+        if (top == NULL)
+        {
+            print("Stack Underflow.");
+        }
+        else
+        {
+            Node *temp = top;
+            top = top->next;
+            delete temp;
+        }
+    }
+
+    // to search
+
+    void search()
+    {
+
+        int data;
+        print("Enter the data to search: ");
+        cin >> data;
+        Node *temp = top;
+
+        while (temp != NULL)
+        {
+            if (temp->data == data)
+            {
+                print("Present");
+                return;
+            }
+            temp = temp->next;
+        }
+
+        print("Not Present");
+    }
+
+    // to update
+
+    void updateWithKey()
```

Comment:
Is updating a particular key is valid operation on stack-ds?

Reply:
No sir it is not.
Removed it.